ros/angel_system_nodes/angel_system_nodes/berkeley_object_detector.py

```diff
@@ -0,0 +1,176 @@
+import os
+
+import numpy as np
+import rclpy
+from rclpy.node import Node
+
+from cv_bridge import CvBridge
+from sensor_msgs.msg import Image
+
+from angel_msgs.msg import ObjectDetection2dSet
+from angel_utils import RateTracker
+
+from angel_system.berkeley.demo import predictor, model
+
+
+BRIDGE = CvBridge()
+
+
+class BerkeleyObjectDetector(Node):
+    """
+    ROS node that runs the berkeley object detector model and outputs
+    `ObjectDetection2dSet` messages.
+    """
+
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+
+        # Inputs
+        self._image_topic = (
+            self.declare_parameter("image_topic", "debug/PVFrames")
+            .get_parameter_value()
+            .string_value
+        )
+        self._model_config = (
+            self.declare_parameter(
+              "model_config",
+              "angel_system/berkeley/configs/MC50-InstanceSegmentation/mask_rcnn_R_101_FPN_1x_demo.yaml")
+            .get_parameter_value()
+            .string_value
+        )
+        self._det_topic = (
+            self.declare_parameter("det_topic", "ObjectDetections")
+            .get_parameter_value()
+            .string_value
+        )
+
+        log = self.get_logger()
+        log.info(f"Image topic: {self._image_topic}")
+        log.info(f"Model Config file: {self._model_config}")
+        log.info(f"Detections topic: {self._det_topic}")
+
+        # Initialize ROS hooks
+        self._subscription = self.create_subscription(
+            Image,
+            self._image_topic,
+            self.listener_callback,
+            1
+        )
+        self._det_publisher = self.create_publisher(
+            ObjectDetection2dSet,
+            self._det_topic,
+            1
+        )
+
+        # Step classifier
+        parser = model.get_parser()
+
+        conf_thr = 0.7
```

Comment:
I would switch this to a parameter. I thought I had already but it might not be pushed or on master